### PR TITLE
CR-1130345 versal: Display board revision and mfg date through xbutil examine…

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
… platform reports.
MFG date info is represented in "Number of minutes from 0:00 hrs 1/1/96 (Represented in GMT)".

Signed-off-by: Rajkumar Rampelli <rampelli@amd.com>

#### Problem solved by the commit
xbutil examine reports doesn't display board revision & mfg date information. So, added changes in XRT to display the same.
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
CR-1130345
#### How problem was solved, alternative solutions (if any) and why they were rejected
Added support in XRT tools to read & display required information.
#### Risks (if any) associated the changes in the commit
NA
#### What has been tested and how, request additional testing if necessary
Reading xbutil examine reports
#### Documentation impact (if any)
NA

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
bash-4.2# xbutil examine -r platform -d

---------------------------------------------------
[0000:d8:00.1] : xilinx_vck5000_gen4x8_qdma_base_2
---------------------------------------------------
Platform
  XSA Name               : xilinx_vck5000_gen4x8_qdma_base_2
  Platform UUID          : 8A0D97AA-4F00-6E56-3818-1EF41366138A
  FPGA Name              :
  JTAG ID Code           : 0x0
  DDR Size               : 0 Bytes
  DDR Count              : 0
  **Revision               : A
  MFG Date               : 0xcd63d7**
  Mig Calibrated         : false
  P2P Status             : not supported

Clocks
  DATA_CLK (Data)        : 300 MHz
  KERNEL_CLK (Kernel)    : 500 MHz

Mac Addresses            : 00:0A:35:0C:F6:3C
                         : 00:0A:35:0C:F6:3D

**Review by chvamshi-xilinx — APPROVED** [review]
Hope you verified in other Alveo cards also.

**Comment by rajkumar-xilinx** [comment]
> Hope you verified in other Alveo cards also.

Hi @chvamshi-xilinx thank you for the review. Yes, I have verified this PR on Alveo card as well.

File: src/runtime_src/core/common/query_requests.h [hunk 1]
```diff
@@ -3073,6 +3074,15 @@ struct hwmon_sdm_fan_presence : request
   get(const device*) const = 0;
 };
 
+struct hwmon_sdm_mfg_date : request
```
Comment (stsoe) [inline src/runtime_src/core/common/query_requests.h:3077]: Please add a brief comment for  all new query requests.
Feel free to comment other ones too :-)

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/common/query_requests.h (hunk 1 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/common/query_requests.h:3078]: Hi Soren, thank you for the review. I added comment for this query.

**Comment by rajkumar-xilinx** [comment]
retest this please

**Comment by dayeh-xilinx** [comment]
retest this please. Jenkins server restart.

File: src/runtime_src/core/tools/common/ReportPlatforms.cpp [hunk 2]
```diff
@@ -67,6 +67,11 @@ ReportPlatforms::writeReport( const xrt_core::device* /*_pDevice*/,
     const boost::property_tree::ptree& pt_board_info = pt_platform.get_child("off_chip_board_info");
     _output << boost::format("  %-23s: %s Bytes\n") % "DDR Size" % pt_board_info.get<std::string>("ddr_size_bytes");
     _output << boost::format("  %-23s: %s \n") % "DDR Count" % pt_board_info.get<std::string>("ddr_count");
+    try {
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportPlatforms.cpp:70]: I like how you used the try/catch block here.

**Review by rozumx — APPROVED** [review]

**Pull request merged**